src/personality/index.js

```diff
@@ -0,0 +1,149 @@
+const fs = require('fs');
+const forEach = require('lodash.foreach');
+const helper = require('../helper');
+const yaml = require('js-yaml');
+
+const robotDictionary = {};
+const humanDictionary = {};
+
+function loadResponseConfig(path) {
+  return yaml.safeLoad(fs.readFileSync(path, 'utf8'));
+}
+
+/*
+  The js-yaml module does not like regular expressions, so we must define them as strings
+  and parse them into Javascript RegExp objects
+*/
+function parseRegex(regexString) {
+  const regexStringParts = regexString.split('/');
+  const regex = regexStringParts[1];
+  const flags = regexStringParts[2];
+
+  return new RegExp(regex, flags);
+}
+
+function createDictionaryEntry(personality) {
+  const dictionaryEntry = Object.assign({}, personality);
+  dictionaryEntry.responses = [];
+  dictionaryEntry.regex = [];
+
+  personality.responses.forEach((response) => {
+    if (helper.isString(response)) {
+      dictionaryEntry.responses.push(response);
+    }
+    else if (helper.isObject(response)) {
+      for (let i = 0; i < response.weight; i++) {
+        dictionaryEntry.responses.push(response.response);
+      }
+    }
+  });
+
+  if (helper.isString(personality.regex)) {
+    dictionaryEntry.regex = [parseRegex(personality.regex)];
+  }
+  else if (Array.isArray(personality.regex)) {
+    dictionaryEntry.regex = personality.regex.map((regex) => parseRegex(regex));
+  }
+
+  return dictionaryEntry;
+}
+
+function initializeDictionaries() {
+  const robotPersonalities = fs.readdirSync(`${__dirname}/robot`);
+  const humanPersonalities = fs.readdirSync(`${__dirname}/human`);
+
+  robotPersonalities.forEach((personality) => {
+    const responseConfig = loadResponseConfig(`${__dirname}/robot/${personality}`);
+    robotDictionary[personality] = createDictionaryEntry(responseConfig);
+  });
+
+  humanPersonalities.forEach((personality) => {
+    const responseConfig = loadResponseConfig(`${__dirname}/human/${personality}`);
+    humanDictionary[personality] = createDictionaryEntry(responseConfig);
+  });
+}
+
+/*
+  This function defines the values available for replacement during postprcessing
+*/
+function createPostprocessMap(match, username) {
+  return Object.assign(
+    { username: username },
+    match !== undefined && { adjective: match.toLowerCase() },
+    match !== undefined && { ADJECTIVE: match.toLowerCase() }
```

Comment:
I think this one should be `match.toUpperCase()` for "I'M A{{ADJECTIVE}} BOT"

Reply:
Wow... Nice catch.

By the way, the reason I'm using `Object.assign` here is because the resulting object, when `match` is undefined, looks like:
```
{ username: 'foobar' }
```

Had I done something like
```
return {
    username: username,
    adjective: match !== undefined ? adjective.toLowerCase() : undefined,
    ADJECTIVE: match !== undefined ? adjective.toUpperCase() : undefined
};
```

the result would look like:
```
{
    username: 'foobar',
    adjective: undefined,
    ADJECTIVE: undefined
}
```

I didn't want those two additional properties to exist at all even if their values would be `undefined`.